test/unit/services/test_system.py

```diff
@@ -2,284 +2,374 @@
+    _expected_systems: MagicMock
+    _obtained_systems: MagicMock
+
+    def mock_list(self):
+        """Mocks repo methods appropriately to test the 'list' service method"""
+
+        # Simply a return currently, so no need to use actual data
+        self._expected_systems = MagicMock()
+        self.test_helpers.mock_list(self.mock_system_repository, self._expected_systems)
+
+    def call_list(self, parent_id: Optional[str]):
+        """Calls the SystemService `list` method"""
+
+        self._parent_id_filter = parent_id
+        self._obtained_systems = self.system_service.list(parent_id)
+
+    def check_list_success(self):
+        """Checks that a prior call to `call_list` worked as expected"""
+
+        self.mock_system_repository.list.assert_called_once_with(self._parent_id_filter)
+
+        assert self._obtained_systems == self._expected_systems
+
+
+class TestList(ListDSL):
+    """Tests for getting a System"""
+
+    def test_list(self):
+        """Test listing Systems"""
+
+        self.mock_list()
+        self.call_list(str(ObjectId()))
+        self.check_list_success()
+
+
+class UpdateDSL(SystemServiceDSL):
+    """Base class for update tests"""
+
+    _stored_system: Optional[SystemOut]
+    _system_patch: SystemPatchSchema
+    _expected_system_in: SystemIn
+    _expected_system_out: MagicMock
+    _updated_system_id: str
+    _updated_system: MagicMock
+    _update_exception: pytest.ExceptionInfo
+
+    def mock_update(self, system_id: str, system_patch_data: dict, stored_system_post_data: Optional[dict]):
+        """Mocks repository methods appropriately to test the 'update' service method
+
+        :param system_id: ID of the system that will be obtained
+        :param system_patch_data: Dictionary containing the patch data as would be required for a
+                                  SystemPatchSchema (i.e. no id, code or created and modified times required)
+        :param stored_system_post_data: Dictionary containing the system data for the existing stored System
+                                        as would be required for a SystemPostSchema (i.e. no id, code or created and
+                                        modified times required)
+        """
+
+        # Stored system
+        self._stored_system = (
+            SystemOut(
+                **SystemIn(
+                    **stored_system_post_data, code=utils.generate_code(stored_system_post_data["name"], "system")
+                ).model_dump(),
+                id=CustomObjectId(system_id),
+            )
+            if stored_system_post_data
+            else None
+        )
+        self.test_helpers.mock_get(self.mock_system_repository, self._stored_system)
+
+        # Patch schema
+        self._system_patch = SystemPatchSchema(**system_patch_data)
+
+        # Updated system
+        self._expected_system_out = MagicMock()
+        self.test_helpers.mock_update(self.mock_system_repository, self._expected_system_out)
```

Comment:
I am not sure this is necessary - I could equally use self.mock_system_repository.get.return_value in the assert later instead of storing this which would make it a bit shorter, although similar to a lot of things in this test file, right now the service layer doesn't do much, later we may need to mock like this with realistic data if there was processing after it to test. This may also be more readable here.